elide-graphql/src/main/java/com/yahoo/elide/graphql/QueryRunner.java

```diff
@@ -0,0 +1,273 @@
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.graphql;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.ElideResponse;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.ErrorObjects;
+import com.yahoo.elide.core.HttpStatus;
+import com.yahoo.elide.core.exceptions.CustomErrorException;
+import com.yahoo.elide.core.exceptions.HttpStatusException;
+import com.yahoo.elide.core.exceptions.InvalidEntityBodyException;
+import com.yahoo.elide.core.exceptions.TransactionException;
+import com.yahoo.elide.security.User;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.owasp.encoder.Encode;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.GraphQLError;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+/**
+ * Entry point for REST endpoints to execute GraphQL queries.
+ */
+@Slf4j
+public class QueryRunner {
+    private final Elide elide;
+    private GraphQL api;
+
+    private static final String QUERY = "query";
+    private static final String OPERATION_NAME = "operationName";
+    private static final String VARIABLES = "variables";
+    private static final String MUTATION = "mutation";
+
+    public QueryRunner(Elide elide) {
+        this.elide = elide;
+
+        PersistentResourceFetcher fetcher = new PersistentResourceFetcher(elide.getElideSettings());
+        ModelBuilder builder = new ModelBuilder(elide.getElideSettings().getDictionary(), fetcher);
+
+        this.api = new GraphQL(builder.build());
+
+        // TODO - add serializers to allow for custom handling of ExecutionResult and GraphQLError objects
+        GraphQLErrorSerializer errorSerializer =
+                new GraphQLErrorSerializer(elide.getElideSettings().isEncodeErrorResponses());
+        SimpleModule module = new SimpleModule("ExecutionResultSerializer", Version.unknownVersion());
+        module.addSerializer(ExecutionResult.class, new ExecutionResultSerializer(errorSerializer));
+        module.addSerializer(GraphQLError.class, errorSerializer);
+        elide.getElideSettings().getMapper().getObjectMapper().registerModule(module);
+    }
+
+    public ElideResponse post(String graphQLDocument, Object user) {
```

Comment:
Not particularly relavent to this PR since it didn't exist before, but it'd be nice to have some javadoc in the main functions of this class describing the operations they perform.

Reply:
👍